### PR TITLE
feat(RHF, sign-in): config RHF and creat UI SignIn

### PR DESCRIPTION
**Pull request by huan206**

### CODE REVIEW

File: src/components/RHF/FormProvider.types.tsx [hunk 1]
```diff
@@ -0,0 +1,20 @@
+import React, { FormEventHandler,ReactNode } from 'react';
```
Comment (itsdeekha) [inline src/components/RHF/FormProvider.types.tsx:1]: The file name should be `FormProvider` instead.
The same with the other files.

File: src/components/RHF/FormProvider.types.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import { FormProvider as RHFormProvider } from 'react-hook-form';
+
+// ----------------------------------------------------------------------
+
+export interface FormProviderProps {
+  children: ReactNode;
+  onSubmit: FormEventHandler<HTMLFormElement>;
+  methods: any;
```
Comment (itsdeekha) [inline src/components/RHF/FormProvider.types.tsx:9]: The type of `methods` must be `UseFormReturn<T>`.

File: src/components/RHF/RHFSelect.types.tsx [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+import { ReactNode } from 'react';
+import { Controller,useFormContext } from 'react-hook-form';
+
+// ----------------------------------------------------------------------
+
+interface RHFSelectProps {
+  name: string;
+  children: ReactNode;
+
+  [key: string]: any;
```
Comment (itsdeekha) [inline src/components/RHF/RHFSelect.types.tsx:11]: Remove all index signatures.

File: src/components/RHF/RHFSelect.types.tsx [hunk 4]
```diff
@@ -0,0 +1,36 @@
+import { TextField } from '@mui/material';
+import { ReactNode } from 'react';
+import { Controller,useFormContext } from 'react-hook-form';
+
+// ----------------------------------------------------------------------
+
+interface RHFSelectProps {
+  name: string;
+  children: ReactNode;
+
```
Comment (Phat0191) [inline src/components/RHF/RHFSelect.types.tsx:10]: delete this line

File: src/components/RHF/RHFSwitch.types.tsx [hunk 5]
```diff
@@ -0,0 +1,42 @@
+// form
+// @mui
+import { alpha, FormControlLabel, styled,Switch } from '@mui/material';
+import { Controller,useFormContext } from 'react-hook-form';
+
+// ----------------------------------------------------------------------
+
+interface RHFSwitchProps {
+  name: string;
+
```
Comment (Phat0191) [inline src/components/RHF/RHFSwitch.types.tsx:10]: the same

File: src/modules/auth/sign-in/index.tsx [hunk 6]
```diff
@@ -1,61 +1,138 @@
...
+                sx={{ backgroundColor: '#5151E5' }}
+              >
+                <Typography variant='subtitle1'>Sign In</Typography>
+              </LoadingButton>
+            </FormProvider>
+          </Grid>
+          <Grid item md={6} sx={{ display: { xs: 'none', md: 'block' }, pl: 2 }}>
+            <Box
+              style={{
+                backgroundImage: `url(${'http://localhost:3000/assets/images/Introducing-3.png'})`,
```
Comment (itsdeekha) [inline src/modules/auth/sign-in/index.tsx:125]: Using `window.location.origin` instead. `${window.location.origin}/assets/images/Introducing-3.png`

File: src/modules/auth/sign-in/index.tsx [hunk 7]
```diff
@@ -1,61 +1,138 @@
...
   const [isShowPassword, setIsShowPassword] = useState<boolean>(false);
+  interface Login {
+    username: string;
+    password: string;
+    remember: boolean;
+  }
+  const LoginSchema = Yup.object().shape({
+    username: Yup.string().required('Username is required').max(100, 'Username too Long!'),
+    password: Yup.string()
+      .required('Password is required')
```
Comment (itsdeekha) [inline src/modules/auth/sign-in/index.tsx:32]: using `i18n` (locales) for translation purposes.

File: src/modules/auth/sign-in/index.tsx [hunk 8]
```diff
@@ -1,61 +1,138 @@
...
-      </LoadingButton>
-    </Container>
+              <LoadingButton
+                fullWidth
+                size='large'
+                type='submit'
+                variant='contained'
+                sx={{ backgroundColor: '#5151E5' }}
+              >
+                <Typography variant='subtitle1'>Sign In</Typography>
```
Comment (itsdeekha) [inline src/modules/auth/sign-in/index.tsx:118]: using `i18n` (locales) for translation purposes.

**Pull request closed without merging**